### PR TITLE
Feat: Refactor risk functions

### PR DESCRIPTION
**Pull request by StanislawMazowieckiAriane**

### CODE REVIEW

**Review by michalrozekariane — APPROVED** [review]

File: src/types/risk.ts [hunk 1]
```diff
@@ -48,15 +48,24 @@ export interface RiskLevels {
...
 export type RiskLevelTypes = keyof RiskLevels;
 
-export interface Metadata {
-  [key: string]: string;
-  supply_type: string;
-  supply_key: string;
+export type RiskScoreFactors = {
+  [key in KeyTypes]?: number;
+} & {
+  supply_type?: 'INFINITE' | 'FINITE';
```
Comment (BartoszSolkaArianelabs) [inline src/types/risk.ts:54]: all factors should be numbers IMO.

This should be removed.

**Review by BartoszSolkaArianelabs — CHANGES_REQUESTED** [review]

File: src/types/risk.ts (hunk 1 above)
Comment (StanislawMazowieckiAriane) [reply src/types/risk.ts]: Done

**Review by BartoszSolkaArianelabs — APPROVED** [review]

File: src/risk/index.ts [hunk 2]
```diff
@@ -103,37 +104,44 @@ const calculateRiskScoreFromTokenId = async ({
 
   const { data: metadata } = await axios.get<Metadata>(uri);
 
-  const riskScore = calculateRiskScore(metadata, customWeights);
+  const { riskScore, riskScoreFactors } = calculateRiskScore(metadata, customWeights);
   const riskLevel = calculateRiskLevel({ score: riskScore, customRiskLevels });
 
   return {
     riskScore,
     riskLevel,
```
Comment (norbert-kulus-arianelabs) [inline src/risk/index.ts:112]: the type of ``riskLevel`` could be specifically ``'NORISK' | 'LOW' | 'MEDIUM' | 'HIGH'`` instead of just ``string``

Comment (StanislawMazowieckiAriane) [reply src/risk/index.ts:112]: Done

**Pull request merged**